hnn_core/network.py

```diff
@@ -219,47 +293,443 @@ def __repr__(self):
+
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        t0 : float
+            Start time of Poisson-distributed spike train
+        T : float
+            End time of spike train
+        rate_constants : dict of floats
+            Rate constant (lambda) of renewal-process generating the samples.
+            Dict keys are the cell names targeted.
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        weights_nmda : dict or None
+            Synaptic weights of NMDA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            delays within the simulated column
+        dispersion_time : float or dict
+            Synaptic delay at the column origin, dispersed laterally as a
+            function of the space_constant. If float, applies to all target
+            cell types. Use dict, to create dispersion->cell mapping.
+        seedcore : int
+            Optional initial seed for random number generator (default: 42).
+            Each artificial drive cell has seed = seedcore + gid
+        """
+        if (t0 is None or T is None or rate_constants is None):
+            raise ValueError('t0, T and rate_constants must all be defined'
+                             f', got {t0}, {T}, {rate_constants}')
+        if T < 0.:
+            raise ValueError('End time of Poisson input cannot be negative')
+        tstop = self.cell_response.times[-1]
+        if T > tstop:
+            raise ValueError(f'End time of Poisson drive cannot exceed '
+                             f'simulation end time {tstop}. Got {T}.')
+        duration = T - t0
+        if duration < 0.:
+            raise ValueError('Duration of Poisson drive cannot be negative')
+
+        if not isinstance(rate_constants, dict):
+            raise ValueError('rate_constants must be a dict of floats')
+        if not set(rate_constants.keys()).issubset(self.cellname_list):
+            raise ValueError(
+                f"Rate constants should be defined for {self.cellname_list}. "
+                f"Got {list(rate_constants.keys())}")
+
+        drive = NetworkDrive()
+        drive['type'] = 'poisson'
+        drive['cell_specific'] = True
+        drive['seedcore'] = seedcore
+
+        drive['dynamics'] = dict(t0=t0, T=T, rate_constants=rate_constants)
+        drive['events'] = []
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, dispersion_time)
+
+    def add_bursty_drive(self, name, *,
+                         distribution=None, t0=None, sigma_t0=None, T=None,
```

Comment:
hummm ... I don't get why you need to add a default `None` here. We can force keyword arguments without having to do a default. See:

```py
def product(x, *, y): 
      return x * y

product(2, y=3) 
```

Reply:
ah!